mkosi/__init__.py

```diff
@@ -789,23 +789,27 @@ def sign_efi_binary(context: Context, input: Path, output: Path) -> None:
         find_binary("sbsign", root=context.config.tools()) is not None
     ):
         with open(output, "wb") as f:
-            run(
-                [
+            cmd = [
                     "sbsign",
-                    "--key", context.config.secure_boot_key,
                     "--cert", context.config.secure_boot_certificate,
                     "--output", "/dev/stdout",
-                    input,
-                ],
-                stdout=f,
-                sandbox=context.sandbox(
-                    options=[
-                        "--ro-bind", context.config.secure_boot_key, context.config.secure_boot_key,
+                ]
+            options = [
                         "--ro-bind", context.config.secure_boot_certificate, context.config.secure_boot_certificate,
                         "--ro-bind", input, input,
                     ]
-                ),
-            )
+            if context.config.secure_boot_key_uri:
+                cmd += [
+                        "--engine", context.config.secure_boot_key_uri[:context.config.secure_boot_key_uri.find(":")],
+                        "--key", context.config.secure_boot_key_uri,
```

Comment:
That's the whole thing with what's before the colon?

Reply:
yes, this needs to include the prefix, pkcs11:somethingsomething

Reply:
One annoying thing is that if a uri used that takes a path this will fail as we won't mount it into the sandbox. But let's just require users to use the other option for that for now.